pkg/controllers/dynakube/extension/service.go

```diff
@@ -0,0 +1,90 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/service"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (r *reconciler) reconcileService(ctx context.Context) error {
+	if !r.dk.PrometheusEnabled() {
+		if meta.FindStatusCondition(*r.dk.Conditions(), extensionsServiceConditionType) == nil {
+			return nil
+		}
+		defer meta.RemoveStatusCondition(r.dk.Conditions(), extensionsServiceConditionType)
+
+		svc, err := r.buildService()
+		if err != nil {
+			log.Error(err, "could not build service during cleanup")
+
+			return err
+		}
+
+		err = service.Query(r.client, r.apiReader, log).Delete(ctx, svc)
+
+		if err != nil {
+			log.Error(err, "failed to clean up extension service")
+
+			return nil
+		}
+
+		return nil
+	}
+
+	return r.createOrUpdateService(ctx)
+}
+
+func (r *reconciler) createOrUpdateService(ctx context.Context) error {
+	newService, err := r.buildService()
+	if err != nil {
+		conditions.SetServiceGenFailed(r.dk.Conditions(), extensionsServiceConditionType, err)
+
+		return err
+	}
+
+	_, err = service.Query(r.client, r.apiReader, log).CreateOrUpdate(ctx, newService)
+	if err != nil {
+		log.Info("failed to create/update extension service")
+		conditions.SetKubeApiError(r.dk.Conditions(), extensionsServiceConditionType, err)
+
+		return err
+	}
+
+	conditions.SetServiceCreated(r.dk.Conditions(), extensionsServiceConditionType, r.buildServiceName())
```

Comment:
the message is 
```
 message: dynakube-extensions-controller created 
 reason: ServiceCreated
```
what do you think about?
```
 message: dynakube-extensions-controller service created 
```

Reply:
@aorcholski 
TBH I don't like this idea because we have a convention: object name + createdSuffix:
```
func appendCreatedSuffix(name string) string {
	return fmt.Sprintf("%s %s", name, createdSuffix)
}
```

if one adds `-secret` to object name it's okey ... but I don't think I need to change message to align with it.

Reply:
it has reason and type to highlight 2 times it's service:
```
  {
    "lastTransitionTime": "2024-07-23T07:45:59Z",
    "message": "dynakube-extensions-controller created",
    "reason": "ServiceCreated",
    "status": "True",
    "type": "ExtensionsService"
  },
```
cc @0sewa0 @albertogdd need 3rd opinion on it.

Reply:
you can just remove the `service` part and say 
```
 message: dynakube-extensions-controller created 
```

Reply:
@0sewa0  that's how it works currently. 🤔

Reply:
oh, can't read